src/workerd/api/node/tests/tls-nodejs-test.js

```diff
@@ -0,0 +1,210 @@
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+import tls from 'node:tls';
+import { fail, strictEqual, ok } from 'node:assert';
+import { getEventListeners, once } from 'node:events';
+
+// Tests are taken from
+// https://github.com/nodejs/node/blob/304743655d5236c2edc39094336ee2667600b684/test/parallel/test-tls-connect-abort-controller.js
+export const tlsConnectAbortController = {
+  async test() {
+    // Our tests differ from Node.js
+    // We don't check for abortSignal listener count because it's not supported.
+    const connectOptions = (signal) => ({
+      port: 8888,
+      host: 'localhost',
+      signal,
+    });
+
+    const assertAbort = async (socket, testName) => {
+      try {
+        await once(socket, 'close');
+        fail(`close ${testName} should have thrown`);
+      } catch (err) {
+        strictEqual(err.name, 'AbortError');
+      }
+    };
+
+    async function postAbort() {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const socket = tls.connect(connectOptions(signal));
+      ac.abort();
+      await assertAbort(socket, 'postAbort');
+    }
+
+    async function preAbort() {
+      const ac = new AbortController();
+      const { signal } = ac;
+      ac.abort();
+      const socket = tls.connect(connectOptions(signal));
+      await assertAbort(socket, 'preAbort');
+    }
+
+    async function tickAbort() {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const socket = tls.connect(connectOptions(signal));
+      setImmediate(() => ac.abort());
+      await assertAbort(socket, 'tickAbort');
+    }
+
+    async function testConstructor() {
+      const ac = new AbortController();
+      const { signal } = ac;
+      ac.abort();
+      const socket = new tls.TLSSocket(undefined, connectOptions(signal));
+      await assertAbort(socket, 'testConstructor');
+    }
+
+    async function testConstructorPost() {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const socket = new tls.TLSSocket(undefined, connectOptions(signal));
+      ac.abort();
+      await assertAbort(socket, 'testConstructorPost');
+    }
+
+    async function testConstructorPostTick() {
+      const ac = new AbortController();
+      const { signal } = ac;
+      const socket = new tls.TLSSocket(undefined, connectOptions(signal));
+      setImmediate(() => ac.abort());
+      await assertAbort(socket, 'testConstructorPostTick');
+    }
+
+    await postAbort();
+    await preAbort();
+    await tickAbort();
+    await testConstructor();
+    await testConstructorPost();
+    await testConstructorPostTick();
```

Comment:
I'd generally prefer these to be their own individual tests

Reply:
This is directly from node.js tests. I deliberately didn't update them to follow up for changes easily. I can split them if you insist.

Reply:
Node.js tests are awful. There's an opportunity to improve both there and here. Not going to insist on it tho

Reply:
Resolve this for now, and I'll try to open a PR to node.js for this soon.